python/tank/deploy/descriptor.py

```diff
@@ -12,551 +12,43 @@
-
-    The cache keys based on the bundle storage root path and
-    the locator dictionary.
-    """
-    _instances = dict()
-
-    def __new__(cls, pc_path, bundle_cache_root, location_dict, *args, **kwargs):
-        """
-        Handles caching of descriptors.
-
-        Executed prior to __init__ being executed.
-
-        Since all our normal descriptors are immutable - they represent a specific,
-        read only and cached version of an app, engine or framework on disk, we can
-        also cache their wrapper objects.
-
-        :param pc_path: Path to pipeline configuration
-        :param bundle_cache_root: Root location for bundle cache
-        :param location_dict: Location dictionary describing the bundle
-        :return: Descriptor instance
-        """
-        instance_cache = cls._instances
-
-        # The cache is keyed based on the location dict and the bundle install root
-        cache_key = (bundle_cache_root, str(location_dict))
-
-        # Instantiate and cache if we need to, otherwise just return what we
-        # already have stored away.
-        if cache_key not in instance_cache:
-            # If the bundle install path isn't in the cache, then we are
-            # starting fresh. Otherwise, check to see if the app (by name)
-            # is cached, and if not initialize its specific cache. After
-            # that we instantiate and store by version.
-            instance_cache[cache_key] = super(VersionedSingletonDescriptor, cls).__new__(
-                cls,
-                pc_path,
-                bundle_cache_root,
-                location_dict,
-                *args, **kwargs
-            )
-
-        return instance_cache[cache_key]
-
-    def __eq__(self, other):
-        # Since this is a singleton descriptor, we can check
-        # to see if it's a reference to the same object.
-        return self is other
-
-    def __ne__(self, other):
-        return self is not other
-
-
-################################################################################################
-# factory method for app descriptors
-
-
-def get_from_location(app_or_engine, pipeline_config, location_dict):
-    """
-    Factory method.
-
-    :param app_or_engine: Either AppDescriptor.APP AppDescriptor.ENGINE or FRAMEWORK
-    :param pipeline_config: Pipeline Configuration Object
-    :param location_dict: A tank location dict
-    :returns: an AppDescriptor object
-    """
-
-    return get_from_location_and_paths(
-        app_or_engine,
-        pipeline_config.get_path(),
-        pipeline_config.get_bundles_location(),
-        location_dict
-    )
-
 
 def get_from_location_and_paths(app_or_engine, pc_path, bundle_install_path, location_dict):
     """
     Factory method.
 
-    :param app_or_engine: Either AppDescriptor.APP AppDescriptor.ENGINE or FRAMEWORK
-    :param pc_path: Path to the root of the pipeline configuration
+    LEGACY - Use create_descriptor instead.
```

Comment:
Yep yep - this is *exactly* why i kept this. Generally I have been trying to be super mindful about this, but as far as my grep-fu would allow, this is the only place i found where we were calling directly, so not too bad, in the grand scheme of things! I'll add a comment about it.